modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java

```diff
@@ -144,33 +144,55 @@ public ConfigurationInstance get(Event event) {
+      updateUsageStatistic(configuration);
+      return configuration;
+    } catch (MuleRuntimeException e) {
+      throw e;
+    } catch (RuntimeException e) {
+      if (e.getCause() instanceof MuleException) {
+        throw (MuleException) e.getCause();
+      }
+      throw e;
+    } finally {
+      cacheReadLock.unlock();
+    }
+  }
+
+  private void updateUsageStatistic(ConfigurationInstance configuration) {
+    MutableConfigurationStats stats = (MutableConfigurationStats) configuration.getStatistics();
+    stats.updateLastUsed();
   }
 
-  private ConfigurationInstance createConfiguration(Pair<ResolverSetResult, ResolverSetResult> values, CoreEvent event)
-      throws MuleException {
+  private ConfigurationInstance createConfiguration(Pair<ResolverSetResult, ResolverSetResult> values, CoreEvent event) {
     assertNotStopping(muleContext, "Mule is shutting down... Cannot create new dynamic configurations");
 
     ConfigurationInstance configuration;
     ResolverSetResult connectionProviderValues = values.getSecond();
-    if (connectionProviderValues != null) {
-      configuration = configurationInstanceFactory.createConfiguration(getName(),
-                                                                       values.getFirst(),
-                                                                       event,
-                                                                       connectionProviderResolver,
-                                                                       connectionProviderValues);
-    } else {
-      configuration = configurationInstanceFactory.createConfiguration(getName(),
-                                                                       values.getFirst(),
-                                                                       event,
-                                                                       ofNullable(connectionProviderResolver));
+    try {
+      if (connectionProviderValues != null) {
+        configuration = configurationInstanceFactory.createConfiguration(getName(),
+                                                                         values.getFirst(),
+                                                                         event,
+                                                                         connectionProviderResolver,
+                                                                         connectionProviderValues);
+      } else {
+        configuration = configurationInstanceFactory.createConfiguration(getName(),
+                                                                         values.getFirst(),
+                                                                         event,
+                                                                         ofNullable(connectionProviderResolver));
+      }
+      registerConfiguration(configuration);
+      return configuration;
+    } catch (MuleException e) {
```

Comment:
Maybe we can always wrap the exception in a custom one, and then always unwrap it?